api/src/main/java/edu/cornell/mannlib/vitro/webapp/searchengine/base/SearchEngineUtil.java

```diff
@@ -0,0 +1,30 @@
+package edu.cornell.mannlib.vitro.webapp.searchengine.base;
+
+import edu.cornell.mannlib.vitro.webapp.config.ConfigurationProperties;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class SearchEngineUtil {
+
+    private static final Log log = LogFactory.getLog(SearchEngineUtil.class);
+
+    public static String getSearchEngineURLProperty() {
+        ConfigurationProperties config = ConfigurationProperties.getInstance();
+        if (config.getProperty("vitro.local.searchengine.url", "").isEmpty()) {
```

Comment:
If `ConfigurationProperties.getInstance()` returns the DummyConfigurationProperties [getProperty](https://github.com/vivo-project/Vitro/blob/main/api/src/main/java/edu/cornell/mannlib/vitro/webapp/config/DummyConfigurationProperties.java#L24) will return null and isEmpty call will throw a null pointer exception.

Reply:
For the public static method to be safely used, please add appropriate null checks. And if possible, add JavaDocs describing that the method may return null.

Reply:
Good find! I did as you proposed. Also, I included `@Nullable` annotation so it is clear that this method can return `null`.